web/api/v2/create-action/index.ts

```diff
@@ -0,0 +1,194 @@
+import { errorResponse } from "@/api/helpers/errors";
+import {
+  getAPIKeyGraphqlClient,
+  getAPIServiceGraphqlClient,
+} from "@/api/helpers/graphql";
+import { verifyHashedSecret } from "@/api/helpers/utils";
+import { validateRequestSchema } from "@/api/helpers/validate-request-schema";
+import { generateExternalNullifier } from "@/lib/hashing";
+import { ApolloError } from "@apollo/client";
+import { NextRequest, NextResponse } from "next/server";
+import * as yup from "yup";
+import {
+  CreateDynamicActionMutation,
+  getSdk as createDynamicActionSdk,
+} from "./graphql/create-dynamic-action.generated";
+import { getSdk as fetchApiKeySdk } from "./graphql/fetch-api-key.generated";
+
+const createActionBodySchema = yup.object({
+  action: yup.string().strict().required(),
+  name: yup.string().strict().optional().default(""),
+  description: yup.string().strict().optional().default(""),
+  max_verifications: yup.number().strict().optional().default(1),
+});
+
+const createActionParamsSchema = yup.object({
+  app_id: yup.string().strict().required(),
+});
+
+export const POST = async (
+  req: NextRequest,
+  { params: rawParams }: { params: { app_id: string } },
+) => {
+  const api_key = req.headers.get("authorization")?.split(" ")[1];
```

Comment:
One last thing since this will become the base template for our other api endpoints. We should specify that the user does 

Authorization: Bearer `api_token`. I believe here we dont' handle bearer. This seems to be compliant with openapi spec

Reply:
merged